app/src/main/java/at/aau/edu/wizards/gameModel/GameModelRules.kt

```diff
@@ -0,0 +1,211 @@
+package at.aau.edu.wizards.gameModel
+
+class GameModelRules(
+    private val players: ArrayList<GameModelPlayer>,
+    val id: Int,
+    private val cardDealer: GameModelDealer
+) {
+
+    val board = ArrayList<GameModelCard>()
+    private val wins = ArrayList<Int>()
+    var trump: GameModelCard = GameModelCard.NoCard
+        private set
+    private var round = 0
+    var currentPlayer = 0
+        private set
+    private var dealer = 0
+    private var winningPlayer = 0
+    var winningCard: GameModelCard = GameModelCard.NoCard
+
+    fun init() {
+        if (round == 0) {
+            round++
+            for (player in players) {
+                player.dealCards(round)
+            }
+            trump = cardDealer.dealCardInSet()
+            dealer = 0
+            currentPlayer = 0
+            cardDealer.resetSet()
+        }
+    }
+
+    private fun nextRound() {
+        if (round == 10) {
+            return endGame()
+        }
+        for (player in players) {
+            player.score(getAmountWon(player.id))
+        }
+        wins.clear()
+        round++
+        for (player in players) {
+            player.dealCards(round)
+        }
+        trump = cardDealer.dealCardInSet()
+        if (++dealer >= players.size) {
+            dealer = 0
+        }
+        currentPlayer = dealer
+        cardDealer.resetSet()
+    }
+
+
+    private fun endGame() {
+        //TODO
```

Comment:
what about that? Shall we create tasks in backlog for such cases, idk?

Reply:
Well, if we refer to our prototype the score screen would come after, so we should switch to the `Fragment` of the score screen. However since we do this inside the fragment we should communicate to this, which in this case I have not done. So actually this is a bit of an oversight from me.

I will implement a call to `sendMove` of the `GameModel`, that sends some String like: "GameEnded", that we can then catch later on in the Fragment without the need to go back into the Logic.

Reply:
There is also a second TODO, but that is marking the interface between networking and the model and as such needs to be implemented with the caller in mind (game fragment). I do not think it necessary to create a backlog task for this.

No further cases exist.